components/script/dom/permissionstatus.rs

```diff
@@ -0,0 +1,189 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use core::clone::Clone;
+use dom::bindings::cell::DOMRefCell;
+use dom::bindings::codegen::Bindings::BluetoothPermissionResultBinding::BluetoothPermissionDescriptor;
+use dom::bindings::codegen::Bindings::EventHandlerBinding::EventHandlerNonNull;
+use dom::bindings::codegen::Bindings::PermissionStatusBinding::{self, PermissionDescriptor, PermissionName};
+use dom::bindings::codegen::Bindings::PermissionStatusBinding::{PermissionState, PermissionStatusMethods};
+use dom::bindings::error::ErrorResult;
+use dom::bindings::js::Root;
+use dom::bindings::reflector::{DomObject, reflect_dom_object};
+use dom::bindings::trace::JSTraceable;
+use dom::eventtarget::EventTarget;
+use dom::globalscope::GlobalScope;
+use heapsize::HeapSizeOf;
+use js::jsapi::{JS_GetIsSecureContext, JSObject, JSTracer};
+use js::rust::get_object_compartment;
+
+// Enum for storing different type of PermissionDescriptors in the same type.
+#[derive(JSTraceable, HeapSizeOf)]
+pub enum PermissionDescriptorType {
+    // TODO(zakorgy): Finish this list.
```

Comment:
Maybe change this NOTE to `Other types are not implemented, because we only need the bluetooth related one.`